watertap/core/zero_order_costing.py

```diff
@@ -1510,6 +1512,69 @@ def cost_well_field(blk):
         blk.config.flowsheet_costing_block.cost_flow(
             blk.unit_model.electricity[t0], "electricity")
 
+    def cost_nanofiltration(blk):
+        """
+        General method for costing nanofiltration. Costing is carried out
+        using either the general_power_law form or the standard form which
+        computes membrane cost and replacement rate.
+        """
+
+        t0 = blk.flowsheet().time.first()
+
+        # Get parameter dict from database
+        parameter_dict = \
+            blk.unit_model.config.database.get_unit_operation_parameters(
+                blk.unit_model._tech_type,
+                subtype=blk.unit_model.config.process_subtype)
+
+        if (blk.unit_model.config.process_subtype == "default"
```

Comment:
I think I will suggest we make this more explicit, and require the YAML file to define which method to use in the costing parameters, rather than rely on the "subtype". I think this will be more amendable to possible future expansion.

I.e. add a new entry in the cost parameters part of the YAML, then look that up here to determine which method to use.

Secondly, I suggest we break this into a "gateway" method that does the check and then calls out to sub-methods to do the actual implementation (i.e. it calls either `cost_power_law_flow` or a new sub-method (e.g.  `cost_membrane_area`).